src/wtf16.rs

```diff
@@ -0,0 +1,165 @@
+use std::{mem, ops, string::FromUtf16Error};
+
+use serde::{Deserialize, Deserializer};
+use serde_bytes::ByteBuf;
+use wtf8::Wtf8;
+
+use crate::{
+    unicode::{is_lead_surrogate, is_trail_surrogate, combine_surrogate_pair},
+    CodePoint,
+};
+
+#[derive(Clone, Debug, Default, Eq, PartialEq, Hash)]
+pub struct Wtf16(Vec<u16>);
```

Comment:
It seems likely that this will/should get extracted into its own crate (even if not very robust) since it will probably eg "bleed into" `tree-sitter-lint-plugin-eslint-builtin`?